### PR TITLE
Mitsubishi update

### PR DESCRIPTION
**Pull request by RubyBailey**
## Description:

Updating Mitsubishi Climate with the following
- Adds Receiver support
- Adds Fan Modes for 3 and 4 fan speeds as well as Quiet fan mode
- Adds supports_dry and supports_fan_only modes
- Adds Horizontal or Wide Vane default setting
- Adds Vertical Vane default setting

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3886

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by Rai-Rai** [comment]
Hi, sorry that I didn't answer your comment yet. Got a newborn, bit earlier than expected so that took/takes all of our time at the moment

**Comment by RubyBailey** [comment]
@Rai-Rai , no worries, family first. Congrats!
Trust your OK with me submitting these PR's (code and docs) to move forward with integrating the changes that you previously started. Test the code  if / when you have the time.

**Comment by RubyBailey** [comment]
Comment to prevent stale closure of PR

File: components/climate/climate_ir.rst [hunk 1]
```diff
@@ -247,6 +247,42 @@ Known working with:
...
+Configuration variables:
+
+- **set_fan_speeds** (*Optional*): Used to set non-default fan speeds. When not used default is **fan_low** = ``fan1``, **fan_medium** = ``fan2``, **fan_hi** = ``fan3``
+
+  - **fan_low** (*Required*, string)
+  - **fan_medium** (*Required*, string)
+  - **fan_hi** (*Required*, string)
+  - Options are ``fan1``, ``fan2``, ``fan3``, ``fan4``, ``fan5``
+- **horizontal_default** (*Optional*, string): What to default to when the AC unit's horizontal direction is *not* set to swing. Default is ``middle``. Options are: ``left``, ``middle-left``, ``middle``, ``middle-right``, ``right``, ``auto``
+- **vertical_default** (*Optional*, string): What to default to when the AC unit's vertical direction is *not* set to swing. Default is ``middle``. Options are: ``down``, ``mdown``, ``middle``, ``mup``, ``up``, ``auto``
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: As per the changes to be made in code PR
```suggestion
- **vertical_default** (*Optional*, string): What to default to when the AC unit's vertical direction is *not* set to swing. Default is ``middle``. Options are: ``down``, ``middle-down``, ``middle``, ``middle-up``, ``up``, ``auto``
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/climate/climate_ir.rst [hunk 2]
```diff
@@ -247,6 +247,42 @@ Known working with:
...
+
+Additonal configurations available for this platform.
+
+Configuration variables:
+
+- **set_fan_speeds** (*Optional*): Used to set non-default fan speeds. When not used default is **fan_low** = ``fan1``, **fan_medium** = ``fan2``, **fan_hi** = ``fan3``
+
+  - **fan_low** (*Required*, string)
+  - **fan_medium** (*Required*, string)
+  - **fan_hi** (*Required*, string)
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: I forgot this one...
```suggestion
- **set_fan_speeds** (*Optional*): Used to set non-default fan speeds. When not used default is **fan_low** = ``fan1``, **fan_medium** = ``fan2``, **fan_high** = ``fan3``

  - **fan_low** (*Required*, string)
  - **fan_medium** (*Required*, string)
  - **fan_high** (*Required*, string)
```

File: components/climate/climate_ir.rst (hunk 1 above)
Comment (RubyBailey) [reply components/climate/climate_ir.rst]: done

File: components/climate/climate_ir.rst (hunk 2 above)
Comment (RubyBailey) [reply components/climate/climate_ir.rst]: done

**Review by RubyBailey — COMMENTED** [review]
This has all been updated

**Review by jesserockz — APPROVED** [review]

**Pull request merged**